### PR TITLE
Fix importing invalid filenames

### LINKED ISSUE #106: Detect invalid filenames in imports
If we don't support spaces (or any other characters) we should flag files with those names and skip them.

### LINKED ISSUE #158: Ensure we don't skip intermediate images with no base file.

### LINKED ISSUE #177: File Import: Error when importing with empty directories
While doing an import, I was having troubles doing the file import, having the following error:

```
$ vip import files XXX uploads/
-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
Importing files for site:
-- Site: XXXXXXXX (#XXX)
-- Environment: xxxxxxx
-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
Are you sure? yes
Counting files...
Importing 0 files...
Importing 0 files...
Import log: /tmp/import-XXX-1493033105172.log
Import log: /tmp/import-XXX-1493033105172.log
Import log: /tmp/import-XXX-1493033105172.log
Import log: /tmp/import-XXX-1493033105172.log
events.js:160
      throw er; // Unhandled 'error' event
      ^

Error: write after end
    at writeAfterEnd (_stream_writable.js:193:12)
    at WriteStream.Writable.write (_stream_writable.js:244:5)
    at /home/xxx/.node/lib/node_modules/wpvip-cli/build/bin/vip-import.js:104:31
    at /home/xxx/.node/lib/node_modules/wpvip-cli/node_modules/async/lib/async.js:52:16
    at /home/xxx/.node/lib/node_modules/wpvip-cli/node_modules/async/lib/async.js:1209:30
    at /home/xxx/.node/lib/node_modules/wpvip-cli/build/bin/vip-import.js:79:9
    at FSReqWrap.oncomplete (fs.js:123:15)
```

After taking a closer look to the `uploads/` directory, I have noticed that it contains a few empty directories, for example `2016/08` (which was the first directory) was empty. So I ran a bash command to remove all the empty files and tried again:

```shell
find . -type d -empty -delete
```

After removing the empty directories, the file import works perfectly.

### LINKED ISSUE #185: queue.drain fires more than once
queue.drain fires every time the queue is empty -- it doesn't guarantee that there are no files left. If we start the import every time the queue is empty, we can end up running the import many times.

https://github.com/Automattic/vip-cli/blob/master/src/bin/vip-import.js#L222

### PR DESCRIPTION
**Pull request by joshbetz**
Fixes #106 
Fixes #185 
Fixes #177 
Fixes #158 
Also fixes duplicate filenames in the output log.

### CODE REVIEW

**Comment by joshbetz** [comment]
cc @david-binda :)

File: src/bin/vip-import.js [hunk 1]
```diff
@@ -156,6 +156,8 @@ program
 									], function( err, file, stats ) {
 										if ( err ) {
 											return cb( err );
+										} else if ( stats.isSymbolicLink() ) {
+											return cb( new Error( 'Invalid file: symlink' ) );
```
Comment (nickdaugherty) [inline src/bin/vip-import.js]: Will returning an error here stop the entire queue, or does it continue on?

Comment (joshbetz) [inline src/bin/vip-import.js]: No, the callback is in this case is called for _each_ task. https://github.com/caolan/async/blob/v1.5.2/README.md#queue

Comment (nickdaugherty) [inline src/bin/vip-import.js]: Alright, should be fine then.

**Pull request merged**